app/models/user.rb

```diff
@@ -8,8 +8,9 @@ class User < ApplicationRecord
 
   has_many :friendships, foreign_key: :user_id, class_name: 'Friendship'
```

Comment:
### Testing friendship:

Steps:

1. Created two users -> `ryto` and `mike`
2. Ran these commands:

```ruby
friendship = ryto.friendships.build(friend_id: mike.id)

friendship.valid? # returns true

friendship.save

# checking now my friendship variable:
friendship # results in below image
```
![image](https://user-images.githubusercontent.com/6015315/62667555-bd26c100-b9ba-11e9-8828-3d0fb800a058.png)

3. Ok now I checked `mike` 's homepage in the browser and true enough there was a friend request coming from `ryto`. So I clicked that mini icon and I checked again the console after:

```ruby
# checked on my friendship variable
friendship
```
![image](https://user-images.githubusercontent.com/6015315/62667606-0414b680-b9bb-11e9-9ee1-f86f154e6c9f.png)

Now it appears `ryto` and `mike` are friends! Yipee.. And it's all great that in both `mike` 's and `ryto` 's (in the browser) headers I can see they each have 1 friend. 

So what about the console side? I would expect:

Running `ryto.friends` to return something:

```ruby
ryto.friends
```
And yes it indeed showed me `ryto` has a friend:

```
ryto.friends
```
Results in:
![image](https://user-images.githubusercontent.com/6015315/62667722-843b1c00-b9bb-11e9-9896-a2d397e53b80.png)

BUT, `mike.friends` resulted in:

![image](https://user-images.githubusercontent.com/6015315/62667735-93ba6500-b9bb-11e9-8d93-e0dcb3f751b8.png)

With this, my analysis is that a lot of our "business" logic fell into the shoulders of either the 
`friendships_controller` or any associated view that displays friendship related data. 

What we can and should have done is something similar to what this [article](https://smartfunnycool.com/friendships-in-activerecord/) mentions.

Please see my comment on the `models/friendship.rb` and `controllers/friendships_controller.rb` files for further on this.

Ooops, [`friendships_controller.rb`](https://github.com/Ispirett/Facebook-like/blob/7a4be6c2777f0f8ddcbbbf647b97ec9956323b37/app/controllers/friendships_controller.rb#L6) isnt included in this PR so let me comment on it here:

As you will see, the lines

```
# line 8
friendship_exist = Friendship.friendship_exist?(current_user.id,friend_params)

# line 11
if friendship_exist.exists?
      flash[:success] = "This person has already sent you a friend request"
      redirect_to root_path
```
Is a logic that should have been caught in the model (any database logic that isn't a simple `find` or any related method should be in the model).

Reply:
We made the User model as a commentable. However, I don't see any way to make comments on the user. I can surely comment on posts, but not on the user. 

Also, this is a mistake We meant to just  track comment's by the user so it's fixed now.